spec.yml

```diff
@@ -0,0 +1,204 @@
+openapi: "3.0.0"
+info:
+  description: "This is the API definition for the Datenkrake frontend."
+  version: "1.0.0"
+  title: "Datenkrake Backend"
+  contact:
+    email: "leonard@marschke.me"
+    url: "https://bauth.marschke.me/about"
+servers:
+  - url: "https://someserver/v1"
+
+paths:
+  /health:
+    get:
+      summary: Gets service health
+      responses:
+        200:
+          description: Service operational
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/HealthResponse'
+        503:
+          description: Service not operational
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/HealthResponse'
+        default:
+          $ref: '#/components/responses/Error'
+
+  /mailstore:
+    get:
+      summary: Get saved mails
+      description: To get filtered results you can specify query parameters, which get concatenated by AND
+      parameters:
+        - name: id
+          in: query
+          description: Filter for id
+          schema:
+            type: string
+            description: Id generated by POST
+        - name: received_before
+          in: query
+          description: Filter for received date
+          schema:
+            type: string
+            format: datetime
+            description: Time in RFC 3339 format
+            example: "2018-06-10T10:11:46.8328983+02:00"
+        - name: received_after
+          in: query
+          description: Filter for received date
+          schema:
+            type: string
+            format: datetime
+            description: Time in RFC 3339 format
+            example: "2018-06-10T10:11:46.8328983+02:00"
+        - name: mail_from
+          in: query
+          description: Filter for mail sender
+          schema:
+            type: string
+            format: mail
+            description: Mail sender address
+        - name: rcpt_to
+          in: query
+          description: Filter for mail receipient
+          schema:
+            type: string
+            format: mail
+            description: Mail rcpt address
+      responses:
+        200:
+          description: Mail content
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  mail_list:
```

Comment:
Remove indirection layer and use #/components/schemas/MailId as dynamic key.

Reply:
https://github.com/OAI/OpenAPI-Specification/issues/687

We would need this for resolving.